content/article/why-i-hate-json-and-you-can-too-.md

```diff
@@ -0,0 +1,177 @@
+`\uXXXX` notation it represents all
+UTF-8 characters via UTF-16 notation. While that’s fine for applications whose
+strings consist solely of characters from the Basic Multilingual Plane
+(BMP)—Unicode code points below 65,536—it’s awkward for anything
+outside this range. Given the prevalence of emoji characters nowadays,
+it’s not unlikely to need to represent Unicode characters above the BMP
+in a JSON document.
+
+You can do this by encoding the raw UTF-8, but some applications require
+that JSON documents be plain ASCII. In that context we have to encode all
+characters as UTF-16 would represent them. This makes it a bit harder for a
+human to read and write JSON; for example, the “pile-of-poo” emoji
+(💩, U+1f4a9) is encoded as `"\ud83d\udca9"`.
+
+# 3. Human Unfriendliness: Comments
+
+[json.org](https://json.org) describes JSON as “easy for humans to read and
+write.” This is true, but it’s only part of the picture: while it may be easy
+for a human to read and write an isolated JSON document, JSON is an
+unfriendly format for teams of humans to _maintain_.
+
+This is true firstly because JSON disallows comments. While some
+JSON implementations do allow them, these are not mutually consistent; for
+example, CPAN’s popular [JSON::XS](https://metacpan.org/pod/JSON::XS)
+uses `#` rather than `//` for end-of-line comments. Interoperability
+problems can arise when a deviation, however minor, from a global
+standard becomes an in-house standard.
+
+JSON’s lack of support for comments is particularly
+awkward in configuration files, which leaves developers to make choices like:
+
+1. Use a nonstandard JSON variant that allows comments. This builds in a
+dependency either on specific implementations, which defeats the point of
+having a standard format, or on a “normalization” step in the workflow,
+which adds additional maintenance overhead.
+
+2. Alter the data schema to incorporate strings that the application ignores
+but that developers can use to annotate the document.
+
+3. Store comments in a separate file, perhaps alongside the JSON.
+
+4. Forgo comments entirely.
+
+None of these is especially inviting. (The “normalization”
+approach—[Crockford’s erstwhile recommendation](https://archive.is/8FWsA)—seems
+the least offensive.) It would be better to keep things simple by using
+a configuration format that allows comments. (Heh, maybe XML?)
+
+# 4. Human Unfriendliness: Commas
+
+A secondary displeasure that JSON brings is the annoyance at its rejection
+of trailing commas. While modern JavaScript engines now tolerate these,
+the JSON standard still forbids them, which adds additional overhead when
+maintaining JSON documents.
+
+# 5. Inefficiency: Strings
+
+Every C programmer deals with two fundamental ways of encoding strings:
+either delimit them with a “special” character (in C’s case, the NUL byte)
+or transmit their length along with their “body”.
+
+JSON, being derived from a programming language, uses the former method.
```

Comment:
well, other programming languages do the second thing. But, those have been lost to history (or should be). It's not as strong as a connection as you make it, I think.